### PR TITLE
add /api/articles/:article_id endpoint along with error handling. Als…

### PR DESCRIPTION
**Pull request by mundocalrisian**
…o updates endpoints.json

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -52,4 +52,37 @@ describe('APP', () => {
...
+            .then((response) => {
+                expect(response.body.article).toHaveProperty('author', expect.any(String))
+                expect(response.body.article).toHaveProperty('title', expect.any(String))
+                expect(response.body.article).toHaveProperty('article_id', expect.any(Number))
+                expect(response.body.article).toHaveProperty('body', expect.any(String))
+                expect(response.body.article).toHaveProperty('topic', expect.any(String))
+                expect(response.body.article).toHaveProperty('created_at', expect.any(String))
+                expect(response.body.article).toHaveProperty('votes', expect.any(Number))
+                expect(response.body.article).toHaveProperty('article_img_url', expect.any(String))
+        })
```
Comment (nc-ali-abdallah) [inline __tests__/app.test.js:69]: this works fine.
Going forward I would have a look into toMatchObject or objectContaining jest matchers they will allow for more robust testing,

File: __tests__/app.test.js [hunk 2]
```diff
@@ -52,4 +52,37 @@ describe('APP', () => {
...
     });
+    describe('GET /api/articles/:article_id', () => {
+        test('should repsond with a status 200 and an object with the appropriate keys', () => {
+            return request(app)
+            .get('/api/articles/3')
+            .expect(200)
+            .then((response) => {
+                expect(response.body.article).toHaveProperty('author', expect.any(String))
+                expect(response.body.article).toHaveProperty('title', expect.any(String))
+                expect(response.body.article).toHaveProperty('article_id', expect.any(Number))
```
Comment (nc-ali-abdallah) [inline __tests__/app.test.js:63]: When testing the article_id in this case do we expect it to be **_any_** number>

**Review by nc-ali-abdallah — COMMENTED** [review]
Awesome job on this one.

File: __tests__/app.test.js (hunk 2 above)
Comment (mundocalrisian) [reply __tests__/app.test.js:63]: Ah yes, this should just be the same as I'm testing for. I've now changed this.

**Pull request merged**